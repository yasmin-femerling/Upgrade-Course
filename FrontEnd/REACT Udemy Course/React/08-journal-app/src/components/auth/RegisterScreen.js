import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';
import { useDispatch, useSelector } from 'react-redux';
import { removeError, setError } from '../../actions/ui';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';

export const RegisterScreen = () => {

  const dispatch = useDispatch();
  const {msgError} = useSelector( state => state.ui);

  const [formValues, handleInputChange] =  useForm({
    name: 'Yasmin',
    email: 'yas@gmail.com',
    password: '123456',
    password2: '123456'
  });

  const { name, email, password, password2 } = formValues;

  const handleRegister = (e) => {
    e.preventDefault();

    if( isFormValid() ) {
      dispatch(startRegisterWithEmailPasswordName(email,password,name));
    }

  }

  const isFormValid = () => {
    if(name.trim().length === 0) {
      dispatch(setError('Nombre requerido'));
      return false;
    }
    if(!validator.isEmail(email)) {
      dispatch(setError('Correo invalido'));
      return false;
    }
    if(password !== password2 || password.length < 5) {
      dispatch(setError('Contrasenas no coinciden o son menores a 5 caracteres'));
      return false;
    }

    dispatch(removeError());
    return true;
  }

  
  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form onSubmit={handleRegister}>

        {
          msgError && (
            <div className='auth__alert-error'>
             { msgError }
            </div>
          )
        }
          <input className="auth__input" autoComplete="off" value={name} onChange={handleInputChange} text="text" placeholder="Name" name="name" />
          <input className="auth__input" autoComplete="off" value={email} onChange={handleInputChange} text="text" placeholder="Email" name="email" />
          <input className="auth__input" autoComplete="off" value={password} onChange={handleInputChange} text="password" placeholder="Password" name="password" />
          <input className="auth__input" autoComplete="off" value={password2} onChange={handleInputChange} text="password" placeholder="Confirm" name="password2" />

          <button className="btn btn-primary btn-block mb-5" type="submit">Register</button>

          <Link to='/auth/login' className="link">Already Registered</Link>
        </form>
      </>
  )
}
