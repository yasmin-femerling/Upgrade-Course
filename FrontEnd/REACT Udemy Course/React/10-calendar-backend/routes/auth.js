const { Router } = require("express");
const { check } = require("express-validator");
const { validarCampos } = require("../middlewates/validar-campos");
const {
  crearUsuario,
  loginUsuario,
  revalidarToken,
} = require("../controllers/auth");
const { validarJWT } = require("../middlewates/validar-jwt");

const router = Router();

router.post(
  "/new",
  [
    // middlewares
    check("name", "El nombre es obligatorio").not().isEmpty(),
    check("email", "El email es obligatorio").isEmail(),
    check("password", "El password debe contener al menos 6 caracteres").isLength({
      min: 6,
    }),
    validarCampos,
  ],
  crearUsuario
);

router.post(
  "/",
  [
    check("email", "El email es obligatorio").isEmail(),
    check("password", "El password debe contener al menos 6 caracteres").isLength({
      min: 6,
    }),
    validarCampos,
  ],
  loginUsuario
);

router.get("/renew", validarJWT, revalidarToken);

module.exports = router;
