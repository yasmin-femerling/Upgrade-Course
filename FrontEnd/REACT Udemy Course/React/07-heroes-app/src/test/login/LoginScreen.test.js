import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { AuthContext } from "../../auth/authContext";
import { LoginScreen } from "../../components/login/LoginScreen";
import { types } from "../../types/types";

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate
}));

describe('Prueba en <LoginScreen/>', () => { 
    
    const contextValue = {
        user:{
            logged:false
        },
        dispatch: jest.fn(),
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <MemoryRouter>
                <Routes>
                    <Route path="/" element={<LoginScreen/>}/>
                </Routes>
            </MemoryRouter>
        </AuthContext.Provider>
    )

    test('debe hacer match el snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe realizar el dispatch y la navegacion', () => { 
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type:types.login,
            payload: {name:'Yasmin'}
        });

        expect(mockNavigate).toHaveBeenCalledWith('/marvel', {replace:true});
        
        localStorage.setItem('lastPath','/dc');

        handleClick();
        
        expect(mockNavigate).toHaveBeenCalledWith('/dc', {replace:true});

     });
 })