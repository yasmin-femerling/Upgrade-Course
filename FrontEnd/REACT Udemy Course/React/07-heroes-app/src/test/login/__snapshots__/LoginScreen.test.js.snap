// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <LoginScreen/> debe hacer match el snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
